src/ui/pages/orders/tabs/commentsTab.page.ts

```diff
@@ -0,0 +1,78 @@
+import { Tab } from './tab.page';
+import { Locator } from '@playwright/test';
+import { logStep } from 'utils/reporter.utils';
+import { ICommentFromResponse } from 'types/orders.type';
+
+export class CommentsTab extends Tab {
+  readonly uniqueElement = this.page.locator('#comments');
+  readonly commentTextarea = this.uniqueElement.locator('#textareaComments');
+  readonly createButton = this.uniqueElement.locator('#create-comment-btn');
+  readonly commentBlocks = this.uniqueElement.locator('div.mx-3');
+  readonly commentText = (block: Locator) => block.locator('p');
+  readonly commentDate = (block: Locator) => block.locator('div.mt-2 span').nth(1);
+  readonly deleteButton = (block: Locator) => block.locator('button.text-danger');
+
+  @logStep('Fill comment textarea')
+  async fillComment(text: string): Promise<void> {
+    await this.commentTextarea.fill(text);
+  }
+
+  @logStep('Click Create button')
+  async clickCreateButton(): Promise<void> {
+    await this.createButton.click();
+  }
+
+  @logStep('Get comments count')
+  async getCommentsCount(): Promise<number> {
+    return await this.commentBlocks.count();
+  }
+
+  @logStep('Get first comment')
+  async getFirstCommentBlock(): Promise<Locator> {
+    return this.commentBlocks.first();
+  }
+
+  @logStep('Get random comment')
+  async getRandomCommentBlock(): Promise<Locator> {
+    const count = await this.getCommentsCount();
+    if (count === 0) throw new Error('No comments found');
+    const randomIndex = Math.floor(Math.random() * count);
+    return this.commentBlocks.nth(randomIndex);
+  }
+
+  @logStep('Get comment data')
+  async getCommentData(commentBlock: Locator): Promise<ICommentFromResponse> {
+    const [text, createdOn, id] = await Promise.all([
+      this.commentText(commentBlock).innerText(),
+      this.commentDate(commentBlock).innerText(),
+      this.deleteButton(commentBlock).getAttribute('id'),
+    ]);
+    return {
+      _id: id ?? '',
+      text,
+      createdOn,
+    };
+  }
+
+  @logStep('Delete comment')
+  async deleteComment(commentBlock: Locator): Promise<void> {
```

Comment:
Вопрос - в сервисах или тестах получается нужно будет сначала достать локатор комментария, а только потом нажать на кнопку ? 
Насколько это удобно будет 
```suggestion
  async clickDeleteButton(commentBlock: Locator): Promise<void> {
```

Reply:
Не придумала как сделать лучше, если честно, чтобы найти кнопку, относящуюся к конкретному комментарию. Ведь после удаления по-хорошему нужно проверить, что именно этого коммента теперь в списке нет. Как я вижу этот тест: 1) получаю рандомный локатор коммента, 2) получаю его данные, 3) удаляю, предавая в метод delete локатор, 4) проверяю, что в списке стало на один меньше, 5) проверяю, что в списке нет коммента с данными из пункта 2.
Это не считая смоук теста - где 1) создала единственный  коммент, 2) получила его локатор, 3) проверила его данные, 4) передала локатор в метод удаления и удалила.
 
Был вариант сделать для блока комментария отдельный класс, у себя на прошлом проекте я так делала для разных айтемов, но по итогу я тоже туда передавала локатор, но метод внутри этого класса выглядел бы как просто клик по этой кнопке, а кнопка была бы deleteButton =  this.root.locator('button.text-danger').

P.S. Метод переименовала